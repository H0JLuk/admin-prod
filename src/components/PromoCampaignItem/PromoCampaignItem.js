import React, { memo } from 'react';
import PropTypes from 'prop-types';
import styles from './PromoCampaignItem.module.css';
import Button from '../Button/Button';
import ButtonLabels from '../Button/ButtonLables';

function PromoCampaignItem({ webUrl, name, handleStatisticsClick }) {
    return (
        <div className={ styles.promoCampaignItem }>
            <div className={ styles.descrWrapper }>
                <div className={ styles.fieldsWrapper }>
                    <p><b>Название:</b> { `"${ name }"` }</p>
                    { webUrl &&
                        <p><b>URL:</b> { `"${ webUrl }"` }</p>
                    }
                </div>
                <div className={ styles.promoCampaignItemActions }>
                    <Button
                        type="blue"
                        label={ ButtonLabels.STATISTICS }
                        onClick={ () => handleStatisticsClick() }
                    />
                </div>
            </div>
        </div>
    );
}

PromoCampaignItem.propTypes = {
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    webUrl: PropTypes.string,
    dzoId: PropTypes.number.isRequired,
    promoCodeType: PropTypes.string.isRequired,
    handleStatisticsClick: PropTypes.func.isRequired,
};

export default memo(PromoCampaignItem);