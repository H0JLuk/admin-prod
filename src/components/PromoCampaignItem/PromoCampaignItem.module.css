.description table{
    width: 100% !important;
    border-collapse: collapse;
}

.description table {
   border: #000014;
}
.description table tr td{
   background-color: white;
}
.description table tr th{
    font-weight: bold;
}
