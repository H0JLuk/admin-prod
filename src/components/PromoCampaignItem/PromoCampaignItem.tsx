import React, { memo } from 'react';
import { Descriptions, Checkbox } from 'antd';
import PropTypes from 'prop-types';

import styles from './PromoCampaignItem.module.css';

const columnsNumber = 1;

type PromoCampaignItemProps = {
    id: number;
    webUrl?: string;
    name: string;
    active?: boolean;
    dzoId: number;
    promoCodeType: string;
    type?: string;
    getDzo: (dzoId: number) => string;
};

const PromoCampaignItem: React.FC<PromoCampaignItemProps> = ({
    webUrl,
    name,
    active,
    dzoId,
    promoCodeType,
    type,
    getDzo
}) => (
    <Descriptions column={columnsNumber} bordered className={styles.description}>
        <Descriptions.Item label="Название">{name}</Descriptions.Item>
        <Descriptions.Item label="ДЗО">{getDzo(dzoId)}</Descriptions.Item>
        <Descriptions.Item label="Ссылка на страницу промо-кампании">{webUrl ? webUrl : '-'}</Descriptions.Item>
        <Descriptions.Item label="Тип промо-кампании">{type}</Descriptions.Item>
        <Descriptions.Item label="Тип промокода">{promoCodeType}</Descriptions.Item>
        <Descriptions.Item label="Активна"><Checkbox checked={active} /></Descriptions.Item>
    </Descriptions>
);

PromoCampaignItem.propTypes = {
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    webUrl: PropTypes.string,
    dzoId: PropTypes.number.isRequired,
    promoCodeType: PropTypes.string.isRequired,
    getDzo: PropTypes.func.isRequired,
};

export default memo(PromoCampaignItem);
