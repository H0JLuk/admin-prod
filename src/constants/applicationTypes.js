const applicationTypes = [
    {type: 'IOS'},
    {type: 'ANDROID'},
    {type: 'OTHER'}
];

export default applicationTypes;