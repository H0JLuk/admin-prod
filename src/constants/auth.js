export const LOGIN_TYPE = {
    PASSWORD: 'PASSWORD',
    // SBOL_PRO: 'SBOL_PRO',
    SBER_USER_ID: 'SBER_USER_ID',
};