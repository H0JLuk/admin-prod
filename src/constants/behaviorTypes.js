const behaviorTypes = {
    QR: 'QrBehavior',
    OKKO: 'OkkoBehavior',
    SBER_LOGISTIC: 'SberLogisticBehavior',
    DUMMY: 'DummyBehavior',
    EMPTY: 'Empty', //technical behavior to display empty labels while awaiting server response
    WEB: 'WebOnlyBehavior',
};

export default behaviorTypes;
