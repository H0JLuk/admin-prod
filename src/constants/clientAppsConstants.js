export const APP_MECHANICS = {
    PRESENTS: { value: 'PRESENTS', label: 'Подарки' },
    ECOSYSTEM: { value: 'ECOSYSTEM', label: 'Продукты' },
    PRESENTATION: { value: 'PRESENTATION', label: 'Экскурсия' },
    BUNDLE: { value: 'BUNDLE', label: 'WOW' },
};

export const MECHANICS_CHECKBOXES = Object.keys(APP_MECHANICS).map(key => ({
    value: APP_MECHANICS[key].value,
    label: APP_MECHANICS[key].label,
}));

export const APP_LOGIN_TYPES = [
    { label: 'Логин/Пароль', value: 'PASSWORD' },
    { label: 'СБОЛ ПРО', value: 'SBOL_PRO' },
    { label: 'Ссылка', value: 'DIRECT_LINK' },
];

export const MECHANICS_ERROR = `Необходимо выбрать хотя бы одну обязательную механику (${APP_MECHANICS.PRESENTS.label}, ${APP_MECHANICS.ECOSYSTEM.label} или ${APP_MECHANICS.BUNDLE.label})`;
