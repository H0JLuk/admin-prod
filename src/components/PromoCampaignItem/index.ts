import PromoCampaignItem from './PromoCampaignItem';

export default PromoCampaignItem;
