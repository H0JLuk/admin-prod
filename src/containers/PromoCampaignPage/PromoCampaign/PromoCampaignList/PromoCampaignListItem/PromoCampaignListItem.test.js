import React from 'react';
import { shallow } from 'enzyme';
import PromoCampaignItem from './PromoCampaignListItem';
import { promoCampaignTestData } from '../../../../../../__tests__/constants';


const mockHistoryPush = jest.fn();
jest.mock('react-router-dom', () => ({
    useHistory: () => ({
        push: mockHistoryPush,
    }),
    generatePath: () => ({ path: 'test' }),
}));

describe('<PromoCampaignItem /> test', () => {
    const initialProps = {
        promoCampaign: { ...promoCampaignTestData },
        onDeleteItem: jest.fn(),
        sortable: false,
        matchUrl: 'url',
    };

    const container = shallow(<PromoCampaignItem { ...initialProps } />);

    it('should match snapshot', () => {
        expect(container.html()).toMatchSnapshot();
    });

    it('onCardClick should be called', () => {
        container.find('.cardInfo').simulate('click');
        expect(mockHistoryPush).toHaveBeenCalledTimes(1);
    });

    it('status promo-campaign should be "Активная"', () => {
        expect(container.find('.status').at(1).text()).toEqual('Активная');
    });

    it('status promo-campaign should be "Неактивная"', () => {
        container.setProps({ promoCampaign: { ...initialProps.promoCampaign, active: false } });
        expect(container.find('.status').at(1).text()).toEqual('Неактивная');
    });

    it('should displayed PromoCampaignItemMenu', () => {
        expect(container.find('.cardMenu').text()).toMatch('PromoCampaignItemMenu');
    });

    it('should displayed sortableIcon', () => {
        container.setProps({ sortable: true });
        expect(container.find('.cardMenu').html()).toMatch('sortableIcon');
    });
});
