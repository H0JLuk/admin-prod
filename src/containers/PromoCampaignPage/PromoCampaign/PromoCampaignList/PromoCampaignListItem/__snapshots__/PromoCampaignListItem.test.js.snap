// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PromoCampaignItem /> test should match snapshot 1`] = `"<div class=\\"cardWrapper\\"><div class=\\"cardInfo\\"><div class=\\"cardInfo__header\\"><div class=\\"cardName\\">Промо-кампания Сбермаркет для витрины ВСП</div><div class=\\"subInfo type\\"><div class=\\"subInfo__label\\">Тип</div><div class=\\"subInfo__value\\">Подарки</div></div><div class=\\"subInfo promoCode\\"><div class=\\"subInfo__label\\">Промо-код</div><div class=\\"subInfo__value\\">Персональный</div></div><div class=\\"subInfo startDate\\"><div class=\\"subInfo__label\\">Создано</div><div class=\\"subInfo__value\\">13.02.2021</div></div><div class=\\"subInfo endDate\\"><div class=\\"subInfo__label\\">Окончание</div><div class=\\"subInfo__value\\">08.03.2021</div></div></div><div class=\\"cardInfo__footer\\"><div class=\\"status\\">СберЗдоровье</div><div class=\\"status\\">Активная</div></div></div><div class=\\"cardAction\\"><div class=\\"cardMenu\\"><div class=\\"ant-dropdown-trigger menuButton dropdown\\"><span role=\\"img\\" aria-label=\\"ellipsis\\" class=\\"anticon anticon-ellipsis\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"ellipsis\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z\\"></path></svg></span></div></div></div></div>"`;
